lib/publiccloud/basetest.pm

```diff
@@ -185,6 +185,7 @@ sub _upload_logs {
 
     my @instance_logs = ('/var/log/cloudregister', '/etc/hosts', '/var/log/zypper.log', '/etc/zypp/credentials.d/SCCcredentials');
     for my $instance_log (@instance_logs) {
+        next if ($self->{run_args}->{my_instance}->ssh_script_run("test -f " . $instance_log, quiet => 1));
```

Comment:
I think this line is redundant given that we have `failok` set below.  We should minimize the number of commands run through ssh.

```suggestion
```

Reply:
> if a log file is **NOT present,** in this way we exclude it from the upload that provides failing commands and also we have LESS useless failing boxes in the openqa details web gui, then a cleaner test report. I opt to keep it.

Can you point to an example of such a red box?

This is redundant and running another ssh command just to test if the file exists is wasteful and increases the chance that the test fails if ssh fails.  This test is useless becase we're already using `failok` like this below:

`$self->{run_args}->{my_instance}->upload_log($instance_log, failok => 1, log_name => $instance_log . ".txt");`